tests/concepts/test_craft_torch.py

```diff
@@ -0,0 +1,380 @@
+        assert(crops_u.shape[1] == w.shape[0])
+        
+        # Importance estimation
+        importances = craft.estimate_importance(nb_most_important_concepts=5)
+        print('Checking shape of importances:', importances.shape)
+        assert(len(importances) == number_of_concepts)
+
+        images_u = craft.transform(images_preprocessed)
+        print('Checking shape of images_u:', images_u.shape)
+        assert(images_u.shape[-1] == number_of_concepts)
+
+def test_wrong_layers():
+    """Ensure that Craft complains when the input models are incompatible"""
+
+    input_shapes = [(3, 32, 32)]
+    nb_labels = 3
+
+    for input_shape in input_shapes:
+
+        model = generate_torch_model(input_shape, nb_labels)
+
+        # cut the model in two parts (as explained in the paper)
+        # first part is g(.) our 'input_to_latent' model, second part is h(.) our 'latent_to_logit' model
+        g = nn.Sequential(*list(model.children())[:3])
+        h = lambda x: 2*x
+
+        # Initialize Craft
+        number_of_concepts = 10
+        patch_size = 15
+        with pytest.raises(TypeError):
+            craft = Craft(input_to_latent_model = g,
+                        latent_to_logit_model = h,
+                        number_of_concepts = number_of_concepts,
+                        patch_size = patch_size,
+                        batch_size = 64)
+
+def test_classifier():
+    """ Check the Craft results on a small fake dataset """
+
+    input_shape = (64, 64, 3)
+    nb_labels = 3
+    nb_samples = 200
+
+    # Create a dataset of 'ABC', 'BCD', 'CDE' images
+    x, y, nb_samples, labels_str = generate_txt_images_data(input_shape, nb_labels, nb_samples)
+    x = np.moveaxis(x, -1, 1) # reorder the axis to match torch format
+    x, y = torch.Tensor(x), torch.Tensor(y)
+    print(f'Dataset: {nb_samples} samples generated, {nb_labels} classes: {labels_str}')
+
+    # train a small classifier on the dataset
+    def generate_torch_model(input_shape=(3, 64, 64), output_shape=10):
+        flatten_size = 6*(input_shape[1]-3)*(input_shape[2]-3)
+        model = nn.Sequential(
+            nn.Conv2d(3, 6, kernel_size=(2, 2)),
+            nn.ReLU(),
+            nn.Conv2d(6, 6, kernel_size=(2, 2)),
+            nn.ReLU(),
+            nn.Conv2d(6, 6, kernel_size=(2, 2)),
+            nn.ReLU(),
+            nn.Flatten(1, -1),
+            # nn.Dropout(p=0.2),
+            nn.Linear(flatten_size, output_shape))
+        for layer in model:
+            if isinstance(layer, nn.Conv2d):
+                nn.init.kaiming_uniform_(layer.weight, mode='fan_in', nonlinearity='relu')
+                layer.bias.data.fill_(0.01)
+            elif isinstance(layer, nn.Linear):
+                nn.init.xavier_normal_(layer.weight)
+                layer.bias.data.fill_(0.01)
+        return model
+    model = generate_torch_model((input_shape[-1], *input_shape[0:2]), nb_labels)
+
+    def train_model(model, x, y):
```

Comment:
I hope the model training is not too time-consuming, otherwise, it may slow down all future tests.

Reply:
Now the model is loaded.

Reply:
As `retrain` args will always be false why don't we simply remove it and remove the training loop. Is there any specific reason ? :)